perfkitbenchmarker/providers/gcp/gce_virtual_machine.py

```diff
@@ -283,11 +283,23 @@ def _GenerateCreateCommand(self, ssh_keys_path):
       cmd.flags['machine-type'] = self.machine_type
     cmd.flags['tags'] = 'perfkitbenchmarker'
     cmd.flags['no-restart-on-failure'] = True
-    cmd.flags['metadata-from-file'] = 'sshKeys=%s' % ssh_keys_path
-    metadata = ['owner=%s' % FLAGS.owner]
-    for key, value in self.boot_metadata.iteritems():
-      metadata.append('%s=%s' % (key, value))
-    cmd.flags['metadata'] = ','.join(metadata)
+
+    metadata_from_file = {'sshKeys': ssh_keys_path} if ssh_keys_path else {}
+    for pair in FLAGS.vm_metadata_from_file:
+      key, value = pair.split('=')
+      metadata_from_file.setdefault(key, value)
```

Comment:
Looks like setting --vm_metadata_from_file sshKeys=value will cause the command-line value to be silently dropped. Maybe there should be an error or warning.